relentless/simulate/hoomd.py

```diff
@@ -642,117 +636,98 @@ def __init__(self, add_op):
-    ------
-    TypeError
-        If the specified addition operation is not a NPT integrator.
-
-    """
-    def __init__(self, add_op):
-        if not isinstance(add_op, AddNPTIntegrator):
-            raise TypeError('Addition operation is not AddNPTIntegrator.')
-        super().__init__(add_op)
-
-class AddNVTIntegrator(AddMDIntegrator):
-    r"""NVT integration via Nosé-Hoover thermostat.
+    - NVE integration with optional Berendsen thermostat
+    - NVT integration with Nosé-Hoover thermostat
+    - NPH integration with MTK barostat
+    - NPT integration with Nosé-Hoover thermostat and MTK barostat
 
     Parameters
     ----------
     dt : float
         Time step size for each simulation iteration.
-    tau_T : float
-        Coupling constant for the thermostat.
+    thermostat : :class:`~relentless.simulate.simulate.Thermostat`
+        Thermostat used for integration (defaults to ``None``).
+    barostat : :class:`~relentless.simulate.simulate.Barostat`
+        Barostat used for integration (defaults to ``None``).
     options : kwargs
-        Options used in :class:`hoomd.md.integrate.nvt`.
+        Options used in :class:`hoomd.md.integrate.nve`,
+        :class:`hoomd.md.integrate.berendsen`, :class:`hoomd.md.integrate.nvt`,
+        :class:`hoomd.md.integrate.nph`, or :class:`hoomd.md.integrate.npt`, as
+        appropriate.
 
     """
-    def __init__(self, dt, tau_T, **options):
+    def __init__(self, dt, thermostat=None, barostat=None, **options):
         super().__init__(dt)
-        self.tau_T = tau_T
+        self.thermostat = thermostat
+        self.barostat = barostat
         self.options = options
 
     def __call__(self, sim):
-        """Adds the NVT integrator to the simulation.
+        """Adds the Verlet integrator to the simulation.
 
         Parameters
         ----------
         sim : :class:`~relentless.simulate.simulate.Simulation`
             The simulation object.
 
+        Raises
+        ------
+        TypeError
+            If an appropriate combination of thermostat and barostat is not set.
+
         """
         self.attach_integrator(sim)
         with sim.context:
             all_ = hoomd.group.all()
-            sim[self].integrator = hoomd.md.integrate.nvt(group=all_,
-                                                          kT=sim.ensemble.kT,
-                                                          tau=self.tau_T,
-                                                          **self.options)
+            if self.thermostat is None and self.barostat is None:
+                sim[self].integrator = hoomd.md.integrate.nve(group=all_,
+                                                              **self.options)
+            elif isinstance(self.thermostat, simulate.BerendsenThermostat) and self.barostat is None:
+                sim[self].integrator = hoomd.md.integrate.berendsen(group=all_,
+                                                                    kT=self.thermostat.T,
```

Comment:
In HOOMD, you need to multiply T by kB from the ensemble everywhere you use the input `kT`:
```
kT=sim.ensemble.kB*self.thermostat.T
```